File: 6-kyu/playing-with-passphrases/python/solution.py
def play_pass(s, n):
    ret = ''
    for i, ch in enumerate(s):
        if ch.isalpha():
            ord_ch = ord(ch) + n
            if ord_ch > 90:
                ord_ch -= 26
            str_ch = str(chr(ord_ch))
            str_ch = str_ch.upper() if i % 2 == 0 else str_ch.lower()
            ret += str_ch
        elif ch.isdigit():
            int_ch = 9 - int(ch)
            ret += str(int_ch)
        else:
            ret += ch
    return ret[::-1]

File: 6-kyu/playing-with-passphrases/c++/solution.cpp
#include <algorithm>
#include <string>

class PlayPass
{
  public:
  static std::string playPass(const std::string &s, int n);
};

std::string PlayPass::playPass(const std::string &s, int n)
{
    std::string ret = s;
    for (unsigned i = 0; i < ret.length(); i++) {
        if (std::isalpha(ret[i])) {
            ret[i] += n;
            if (ret[i] > 90) ret[i] -= 26;
            if (i % 2 == 0) ret[i] = std::toupper(ret[i]);
            else ret[i] = std::tolower(ret[i]);
        }
        else if (std::isdigit(ret[i])) {
            ret[i] = (9 - (ret[i] - '0')) + '0';
        }
    }
    std::reverse(ret.begin(), ret.end());
    return ret;
}